pkg/controller/replication/replication_controller_test.go

```diff
@@ -889,3 +889,50 @@ func TestOverlappingRCs(t *testing.T) {
 		}
 	}
 }
+
+func BenchmarkGetPodControllers(b *testing.B) {
+	client := client.NewOrDie(&client.Config{Host: "", GroupVersion: testapi.Default.GroupVersion()})
+	manager := NewReplicationManager(client, controller.NoResyncPeriodFunc, BurstReplicas)
+
+	pods := []api.Pod{}
+	for i := 0; i < 100; i++ {
+		ns := fmt.Sprintf("ns-%d", i)
+		for j := 0; j < 10; j++ {
+			rcName := fmt.Sprintf("rc-%d", j)
+			for k := 0; k < 10; k++ {
+				podName := fmt.Sprintf("pod-%d-%d", j, k)
+				pods = append(pods, api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						Name: podName,
+						Namespace: ns,
+						Labels: map[string]string{"rcName": rcName},
+					},
+				})
+			}
+		}
+	}
+
+	b.ResetTimer()
+
+	for i := 0; i < 100; i++ {
+		ns := fmt.Sprintf("ns-%d", i)
+		for j:= 0; j < 10; j++ {
+			rcName := fmt.Sprintf("rc-%d", j)
+			manager.rcStore.Add(&api.ReplicationController{
```

Comment:
I am not sure. Intuitively, when rc is inserted into store, building index may take some time, so I took those into consideration. I can try to not count the index building time later, to see if there is a big difference.